pkg/autodiscovery/configmgr.go

```diff
@@ -346,22 +346,29 @@ func (cm *reconcilingConfigManager) reconcileService(svcID string) configChanges
 
 	serviceAndADIDs := cm.activeServices[svcID]
 	adIDs := serviceAndADIDs.adIDs // nil slice if service is not defined
+	svc := serviceAndADIDs.svc
 
 	// get the existing resolutions for this service
 	existingResolutions, found := cm.serviceResolutions[svcID]
 	if !found {
 		existingResolutions = map[string]string{}
 	}
 
-	// calculate the expected matching templates by template digest
-	expectedResolutions := map[string]struct{}{}
+	// determine the matching templates by template digest
+	expectedResolutions := map[string]integration.Config{}
 	for _, adID := range adIDs {
 		digests := cm.templatesByADID.get(adID)
 		for _, digest := range digests {
-			expectedResolutions[digest] = struct{}{}
+			tpl := cm.activeConfigs[digest]
+			expectedResolutions[digest] = tpl
 		}
 	}
 
+	// allow the service to filter those templates
+	if svc != nil {
+		svc.FilterTemplates(expectedResolutions)
```

Comment:
It can be, but in that case adIDs is empty as well so expectedResolutions is empty so `cm.resolveTemplateForService` is never called.  I had one tiny comment about that:
```
348     adIDs := serviceAndADIDs.adIDs // nil slice if service is not defined
```
but yeah, it's kinda subtle.  I'll add more comments.

Reply:
cool thanks for clarifying!